### PR TITLE
Remove the split stages implementation of the hybrid image decoder

### PR DESCRIPTION
**Pull request by jantonguirao**
#### Why we need this PR?
*Pick one, remove the rest*
- It removes the implementation of the split stage of the hybrid image decoder since it was no longer used or maintained.

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *Deprecated `split_stages` and `use_chunk_allocator` arguments of `ImageDecoder`*
     *Remove nvjpeg split stages implementation*
     *Removed related tests*
 - Affected modules and functionalities:
     *ImageDecoder*
 - Key points relevant for the review:
     *NA*
 - Validation and testing:
     *NA*
 - Documentation (including examples):
     *Docstring updated*

**JIRA TASK**: *[DALI-1888]*

### CODE REVIEW

**Comment by jantonguirao** [comment]
!build

File: dali/operators/decoder/image_decoder.cc [hunk 1]
```diff
@@ -97,15 +97,23 @@ Otherwise, the threads can be reassigned to any CPU core by the operating system
   .AddOptionalArg("split_stages",
       R"code(Applies **only** to the ``mixed`` backend type.
 
-If True, the operator will be split into two sub-stages: a CPU and GPU one.)code",
+If True, the operator will be split into two sub-stages: a CPU and GPU one.
+
+Warning: This argument is now deprecated, and its value will be ignored.
+)code",
       false)
+  .DeprecateArg("split_stages")  // deprecated in DALI 1.0
```
Comment (awolant) [inline dali/operators/decoder/image_decoder.cc:105]: I think we could just straight up remove it. If there is a time for a breaking change it is 1.0. And no one uses it anyway.

**Review by awolant — APPROVED** [review]

**Review by mzient — APPROVED** [review]
Please search the codebase for occurrences of all deprecated arguments.

File: dali/operators/decoder/image_decoder.cc (hunk 1 above)
Comment (banasraf) [reply dali/operators/decoder/image_decoder.cc:105]: I agree

**Review by banasraf — APPROVED** [review]

**Pull request merged**